src/metrics/summary.rs

```diff
@@ -0,0 +1,164 @@
+//! Module implementing an Open Metrics summary.
+//!
+//! See [`Summary`] for details.
+
+use crate::encoding::{EncodeMetric, MetricEncoder};
+
+use super::{MetricType, TypedMetric};
+use parking_lot::RwLock;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
```

Comment:
`std::time::Instant` is problematic with wasm. Consider using https://crates.io/crates/instant instead.